api/views/author.py

```diff
@@ -225,53 +199,38 @@ def create(self, request, *args, **kwargs):
             friend = CachedAuthor.objects.get(id=friend_id)
 
         # parse any incoming api calls from other nodes
-        if request.get_host() is not settings.FRONTEND_HOST:
-            self.add_request(author, friend)
-            # TODO: check if friend request has been sent already, convert to
-                # friends
-                # self.add_friend(friend, author)
-                # self.add_friend(author, friend)
-                # headers = self.get_success_headers(serializer.data)
-                # return Response(serializer.data, status=status.HTTP_202_ACCEPTED,
-                                # headers=headers)
+        if request.get_host() not in [settings.HOST, settings.FRONTEND_HOST, "testserver"]:
+            author.add_request(friend)
 
         # Otherwise, figureout how to handle the request
         else:
-
             # perform remote calls if necessary
             if friend.is_local() is False:
                 integrator = Integrator.build_from_host(friend["host"])
                 success = integrator.send_friend_request(
                     CachedAuthor(**serializer.data["author"]),
-                    CachedAuthor(**friend)
+                    friend
                 )
 
                 if not success:
                     # TODO: Exception of some sort
                     pass
 
-                # TODO add pending state
-                self.add_following(author, friend)
+                author.follow(friend)
+                author.add_pending(friend)
 
+            # otherwise assume we're operating on two local authors
             else:
-
-                # If friend is already following -> Instant friendship
-                if self.is_following(friend, author) is False:
-                    self.add_following(author, friend)
-
-                # TODO: check if friend request has been sent already, convert to
-                # friends
-                # self.add_friend(friend, author)
-                # self.add_friend(author, friend)
-                # headers = self.get_success_headers(serializer.data)
-                # return Response(serializer.data, status=status.HTTP_202_ACCEPTED,
-                                # headers=headers)
-
-                # Otherwise add resuest to friend
-                self.add_request(friend, author)
+                # if both want to be friends, make it so
+                if author.is_pending_friend(friend):
+                    author.add_friend(friend)
+                    friend.add_friend(author)
+                else:
+                    author.add_pending(friend)
+                    friend.add_request(author)
 
         headers = self.get_success_headers(serializer.data)
-        return Response(serializer.data, status=status.HTTP_201_CREATED,
+        return Response(serializer.data, status=status.HTTP_200_OK,
```

Comment:
Changing it won't break the tests. It's not tested, as I didn't expect that would be the final response.